### DISCUSSION TITLE
Identify OAuth2.0 accounts

### DISCUSSION THREAD

**Discussion by nayakanirudh** [pocketbase/pocketbase#6160] [Q&A] [answered]
Is there a built in way I can identify accounts created via OAuth route? Or I will have to pipe in some info on account creation via createData?

**Comment by ganigeorgiev** [comment, marked as answer]
It depends what PocketBase version you use and from where/how you want to perform the check.

For PocketBase v0.23+:
- Web API: you can send a List API request to the `_externalAuths` system collection as the authenticated user
- Go: [`app.FindAllExternalAuthsByRecord(record)`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.23.11/core#BaseApp.FindAllExternalAuthsByRecord)
- JSVM: [`$app.findAllExternalAuthsByRecord(record)`](https://pocketbase.io/jsvm/functions/_app.findAllExternalAuthsByRecord.html)

For PocketBase <= v0.22:
- Web API: https://pocketbase.io/old/docs/api-records/#list-linked-external-auth-providers
- Go: [`app.Dao().FindAllExternalAuthsByRecord(record)`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.29/daos#Dao.FindAllExternalAuthsByRecord)
- JSVM: [`$app.dao().findAllExternalAuthsByRecord(record)`](https://pocketbase.io/old/jsvm/classes/Dao.html#findAllExternalAuthsByRecord)
